tests/integration/test_shard_names/test.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+from helpers.config_manager import ConfigManager
+
+cluster = ClickHouseCluster(__file__)
+
+node = cluster.add_instance(
+    "node",
+    main_configs=["configs/config_reloader.xml", "configs/cluster.xml"],
+)
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_shard_names(start_cluster):
```

Comment:
FYI
Have moved checks to the integration test, because haven't managed to make them stable in the functional tests.
Because: 
We want to check that `system reload config` query will fail with the `INVALID_SHARD_ID` exception.
* If the config reloader update earlier then `system reload config` will be successful. func test:  
<details><summary>Test</summary>

```
#!/usr/bin/env bash
# Tags: no-fasttest, no-parallel

CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
# shellcheck source=../shell_config.sh
. "$CUR_DIR"/../shell_config.sh

config_reload_name=config_reload_interval.xml
config_incorrect_cluster=${CLICKHOUSE_CONFIG_DIR}/config.d/${config_reload_name}

rm -f $config_incorrect_cluster
$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"

cat > "$config_incorrect_cluster" <<EOL
<clickhouse>
     <remote_servers>
         <cluster>
            <node>
                <host>localhost</host>
                <port>9000</port>
            </node>
            <node>
                <host>localhost</host>
                <port>9000</port>
            </node>
            <incorrect></incorrect> 
         </cluster>
     </remote_servers>
 </clickhouse>
EOL

sleep 10

$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"

rm $config_incorrect_cluster
$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"
```

</details>

* So we need to disable configs auto updates  with `config_reload_interval_ms` opt. But the option suddenly updates only with ConfigReloader, `system reload config` doen't change the value.

<details><summary>Test</summary>

```
#!/usr/bin/env bash
# Tags: no-fasttest, no-parallel

CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
# shellcheck source=../shell_config.sh
. "$CUR_DIR"/../shell_config.sh


config_reload_name=config_reload_interval.xml
config_reload_path=${CLICKHOUSE_CONFIG_DIR}/config.d/${config_reload_name}

config_reload_path=/data/disk1/home/mburdukov/workspace/configs/config.d/${config_reload_name}

rm -f $config_reload_path
$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"

cat > "$config_reload_path" <<EOL
<clickhouse>
  <config_reload_interval_ms>300</config_reload_interval_ms>
</clickhouse>
EOL

$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"
$CLICKHOUSE_CLIENT --query="SELECT value FROM system.server_settings WHERE name='config_reload_interval_ms';" ### `value` == 2000 , assume it must be 300. 


rm $config_reload_path
$CLICKHOUSE_CLIENT --query="SYSTEM RELOAD CONFIG;"

```

</details>